### PR TITLE
Remove constant output in encoder-decoder ONNX models decoder with past 

### PR DESCRIPTION
**Pull request by fxmarty**
Remove the constant output `encoder_last_hidden_state` from decoders.

Breaking change:
* encoder_last_hidden_state is not anymore an output of without/with past decoders ONNX.
* encoder_hidden_states is not anymore an input of without past decoder ONNX, as it is unused (except for t5).

Fixes https://github.com/huggingface/optimum/issues/869

### CODE REVIEW

File: optimum/exporters/onnx/model_patcher.py [hunk 1]
```diff
@@ -126,12 +126,17 @@ def patched_forward(*args, **kwargs):
             # Filter out cross attention past key values
             filterd_outputs = {}
             for name, value in outputs.items():
+                if name != "past_key_values":
+                    if self.real_config._behavior == "decoder" and name == "encoder_last_hidden_state":
+                        # who cares about the encoder outputs in the decoder?
```
Comment (michaelbenayoun) [inline optimum/exporters/onnx/model_patcher.py:131]:
```suggestion
                        # Who cares about the encoder outputs in the decoder?
```

**Review by michaelbenayoun — APPROVED** [review]
Not sure I understand everything here, but LGTM

**Comment by fxmarty** [comment]
Until we support optional inputs, we will need `encoder_last_hidden_state` dummy input (at least in ONNX export validation) so that the merge case works.

**Comment by fxmarty** [comment]
Merging as the failing test is unrelated (https://huggingface.co/hf-internal-testing/tiny-random-DebertaV2Model was updated)

**Pull request merged**